From fd9861fb72e25bf25242551185d0a319305f49bb Mon Sep 17 00:00:00 2001
From: Julan Hsu <julanhsu@google.com>
Date: Tue, 15 Jan 2019 15:28:43 -0800
Subject: [PATCH] FROMGIT: nl80211/mac80211: mesh: add mesh path change count
 to mpath info

Expose path change count to destination in mpath info

Signed-off-by: Julan Hsu <julanhsu@google.com>
Signed-off-by: Johannes Berg <johannes.berg@intel.com>

(cherry-picked from commit 540bbcb930ed2fc9d6a57e0babea00027a7ecc67
git://git.kernel.org/pub/scm/linux/kernel/git/jberg/mac80211-next.git)

BUG=b:117941803
TEST=use mpath dump to verify path change over time

Change-Id: Iefdf2c782fda4545c714e2989a04c5d77b809177
Reviewed-on: https://chromium-review.googlesource.com/1450134
Commit-Ready: ChromeOS CL Exonerator Bot <chromiumos-cl-exonerator@appspot.gserviceaccount.com>
Tested-by: Julan Hsu <julanhsu@google.com>
Reviewed-by: Brian Norris <briannorris@chromium.org>
---
 include/net/cfg80211.h       | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/net/cfg80211.h b/include/net/cfg80211.h
index 330d6678b29a..f84efe908aae 100644
--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -1262,7 +1262,8 @@ enum mpath_info_flags {
 	MPATH_INFO_DISCOVERY_TIMEOUT	= BIT(4),
 	MPATH_INFO_DISCOVERY_RETRIES	= BIT(5),
 	MPATH_INFO_FLAGS		= BIT(6),
-	MPATH_INFO_HOP_COUNT		= BIT(7)
+	MPATH_INFO_HOP_COUNT		= BIT(7),
+	MPATH_INFO_PATH_CHANGE		= BIT(8),
 };
 
 /**
@@ -1283,6 +1284,7 @@ enum mpath_info_flags {
  *	changes, i.e. when a station is added or removed, so that
  *	userspace can tell whether it got a consistent snapshot.
  * @hop_count: hops to destination
+ * @path_change_count: total number of path changes to destination
  */
 struct mpath_info {
 	u32 filled;
@@ -1294,6 +1296,7 @@ struct mpath_info {
 	u8 discovery_retries;
 	u8 flags;
 	u8 hop_count;
+	u32 path_change_count;
 
 	int generation;
 };
-- 
2.21.0.360.g471c308f928-goog

